### PR TITLE
Encode url parameter for html content in hyprlogin

### PR DESCRIPTION
**Pull request by Lakshan-Banneheke**
### Purpose
$subject

### CODE REVIEW

File: identity-apps-core/apps/authentication-portal/src/main/webapp/hyprlogin.jsp [hunk 1]
```diff
@@ -164,9 +164,9 @@
                     pollAuthStatus();
 
                 } else if (status == 'CANCELED' || status == 'FAILED' || status == 'INVALID_REQUEST' || status == 'INVALID_TOKEN') {
-                    handleError(urlParams.get("message"));
+                    handleError('<%= Encode.forHtmlContent(request.getParameter("message")) %>');
```
Comment (Achintha444) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/hyprlogin.jsp:167]:
```suggestion
                    handleError('<%= Encode.forJava(request.getParameter("message")) %>');
```

https://javadoc.io/static/org.owasp.encoder/encoder/1.2.3/org/owasp/encoder/Encode.html#forJava-java.lang.String-

File: identity-apps-core/apps/authentication-portal/src/main/webapp/hyprlogin.jsp [hunk 2]
```diff
@@ -164,9 +164,9 @@
...
                 } else if (status == 'INVALID_USER') {
-                    handleUserError(urlParams.get("message"));
+                    handleUserError('<%= Encode.forHtmlContent(request.getParameter("message")) %>');
```
Comment (Achintha444) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/hyprlogin.jsp:169]:
```suggestion
                    handleUserError('<%= Encode.forJava(request.getParameter("message")) %>');
```

https://javadoc.io/static/org.owasp.encoder/encoder/1.2.3/org/owasp/encoder/Encode.html#forJava-java.lang.String-

File: identity-apps-core/apps/authentication-portal/src/main/webapp/hyprlogin.jsp (hunk 1 above)
Comment (Lakshan-Banneheke) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/hyprlogin.jsp:167]: This is not being encoded to use in Java. It is encoded and then placed in the innerHTML[1]. Hence why Encode.forHtmlContent was used.

[1] https://github.com/Lakshan-Banneheke/identity-apps/blob/master/identity-apps-core/apps/authentication-portal/src/main/webapp/hyprlogin.jsp#L262

File: identity-apps-core/apps/authentication-portal/src/main/webapp/hyprlogin.jsp (hunk 2 above)
Comment (Lakshan-Banneheke) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/hyprlogin.jsp:169]: This is not being encoded to use in Java. It is encoded and then placed in the innerHTML[1]. Hence why Encode.forHtmlContent was used.

[1] https://github.com/Lakshan-Banneheke/identity-apps/blob/master/identity-apps-core/apps/authentication-portal/src/main/webapp/hyprlogin.jsp#L270

File: identity-apps-core/apps/authentication-portal/src/main/webapp/hyprlogin.jsp (hunk 1 above)
Comment (Achintha444) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/hyprlogin.jsp:167]: Ack

File: identity-apps-core/apps/authentication-portal/src/main/webapp/hyprlogin.jsp (hunk 2 above)
Comment (Achintha444) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/hyprlogin.jsp:169]: Ack

**Review by Achintha444 — APPROVED** [review]

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**